File: windows/Liferay.SDK/Service/V73/Segmentsexperimentrel/SegmentsexperimentrelService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Segmentsexperimentrel
{
	public class SegmentsexperimentrelService : ServiceBase
	{
		public SegmentsexperimentrelService(ISession session)
			: base(session)
		{
		}

		public async Task<IEnumerable<dynamic>> GetSegmentsExperimentRelsAsync(long segmentsExperimentId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperimentId", segmentsExperimentId);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperimentrel/get-segments-experiment-rels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<dynamic> AddSegmentsExperimentRelAsync(long segmentsExperimentId, long segmentsExperienceId, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperimentId", segmentsExperimentId);
			_parameters.Add("segmentsExperienceId", segmentsExperienceId);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperimentrel/add-segments-experiment-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateSegmentsExperimentRelAsync(long segmentsExperimentRelId, double split)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperimentRelId", segmentsExperimentRelId);
			_parameters.Add("split", split);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperimentrel/update-segments-experiment-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateSegmentsExperimentRelAsync(long segmentsExperimentRelId, string name, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperimentRelId", segmentsExperimentRelId);
			_parameters.Add("name", name);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperimentrel/update-segments-experiment-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> DeleteSegmentsExperimentRelAsync(long segmentsExperimentRelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperimentRelId", segmentsExperimentRelId);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperimentrel/delete-segments-experiment-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetSegmentsExperimentRelAsync(long segmentsExperimentId, long segmentsExperienceId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("segmentsExperimentId", segmentsExperimentId);
			_parameters.Add("segmentsExperienceId", segmentsExperienceId);

			var _command = new JsonObject()
			{
				{ "/segments.segmentsexperimentrel/get-segments-experiment-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/segmentsexperimentrel/SegmentsexperimentrelService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.segmentsexperimentrel;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class SegmentsexperimentrelService extends BaseService {

	public SegmentsexperimentrelService(Session session) {
		super(session);
	}

	public JSONObject addSegmentsExperimentRel(long segmentsExperimentId, long segmentsExperienceId, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperienceId", segmentsExperienceId);
			_params.put("segmentsExperimentId", segmentsExperimentId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/segments.segmentsexperimentrel/add-segments-experiment-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deleteSegmentsExperimentRel(long segmentsExperimentRelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperimentRelId", segmentsExperimentRelId);

			_command.put("/segments.segmentsexperimentrel/delete-segments-experiment-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getSegmentsExperimentRel(long segmentsExperimentId, long segmentsExperienceId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperienceId", segmentsExperienceId);
			_params.put("segmentsExperimentId", segmentsExperimentId);

			_command.put("/segments.segmentsexperimentrel/get-segments-experiment-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getSegmentsExperimentRels(long segmentsExperimentId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperimentId", segmentsExperimentId);

			_command.put("/segments.segmentsexperimentrel/get-segments-experiment-rels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONObject updateSegmentsExperimentRel(long segmentsExperimentRelId, double split) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("segmentsExperimentRelId", segmentsExperimentRelId);
			_params.put("split", split);

			_command.put("/segments.segmentsexperimentrel/update-segments-experiment-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateSegmentsExperimentRel(long segmentsExperimentRelId, String name, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("name", checkNull(name));
			_params.put("segmentsExperimentRelId", segmentsExperimentRelId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/segments.segmentsexperimentrel/update-segments-experiment-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}